nodejs/eks/cluster.ts

```diff
@@ -184,38 +187,54 @@ export function createCore(name: string, args: ClusterOptions, parent: pulumi.Co
     // Create the VPC CNI management resource.
     const vpcCni = new VpcCni(`${name}-vpc-cni`, kubeconfig, args.vpcCniOptions, { parent: parent });
 
-    // Create the instance role we'll use for worker nodes.
-    let instanceRole: pulumi.Output<aws.iam.Role>;
-    if (args.instanceRole) {
-        instanceRole = pulumi.output(args.instanceRole);
+    let instanceRoleMappings: pulumi.Output<RoleMapping[]>;
+    let instanceProfile = {} as aws.iam.InstanceProfile;
+    if (args.instanceRoles) {
+        // Create role mappings of the instanceRoles specified for aws-auth.
+        const roleMaps: pulumi.Output<RoleMapping[]> = pulumi.output(args.instanceRoles).apply(instanceRoles =>
+            instanceRoles.map(role => createInstanceRoleMapping(role.arn)),
+        );
+        instanceRoleMappings = roleMaps;
     } else {
-        instanceRole = (new ServiceRole(`${name}-instanceRole`, {
-            service: "ec2.amazonaws.com",
-            managedPolicyArns: [
-                "arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy",
-                "arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy",
-                "arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly",
-            ],
-        }, { parent: parent })).role;
-    }
-    const instanceRoleARN = instanceRole.apply(r => r.arn);
+        // Create the instance role we'll use for worker nodes.
+        let instanceRole: pulumi.Output<aws.iam.Role>;
+        if (args.instanceRole) {
+            instanceRole = pulumi.output(args.instanceRole);
+        } else {
```

Comment:
The logic here feels more complex than it needs to be.  There are three cases right - `if (args.instanceRoles)` or `if (args.instanceRole)` or `else`.  And all three branches lead to a list of `instanceRoles`.  And then in all cases we create a `roleMappings`.  

I may be missing something - but this looks like there's a fair bit of duplication and missing symmetry which makes it hard for me to be sure we aren't missing something on one of these branches.